includes/embeds/class-amp-gallery-embed.php

```diff
@@ -171,14 +171,20 @@ public function maybe_override_gallery( $html, $attributes ) {
 					return $attr;
 				};
 
+				$set_link_attribute = static function ( $attributes ) {
+					$attributes['link'] = 'none';
+					return $attributes;
+				};
+
 				remove_filter( 'post_gallery', [ $this, 'maybe_override_gallery' ], 10 );
 				add_filter( 'wp_get_attachment_image_attributes', $add_lightbox_attribute );
+				add_filter( 'shortcode_atts_gallery', $set_link_attribute, PHP_INT_MAX );
 
-				$attributes['link'] = 'none';
-				$html               = gallery_shortcode( $attributes );
+				$html = gallery_shortcode( $attributes );
```

Comment:
When [gallery_shortcode()](https://github.com/WordPress/wordpress-develop/blob/80b4dbf0d0e15a0f5bd6b4c9c243a948c89ce986/src/wp-includes/media.php#L1802) runs, it's possible to filter the `$attributes` with the [shortcode_atts_gallery](https://github.com/WordPress/wordpress-develop/blob/1f5525d2ffd60278018732d3c58f78453cdf57f9/src/wp-includes/shortcodes.php#L583) filter. 

And that can change `$attributes['link'] = 'none';`, causing the lightbox to not work.